Python/Product/PythonTools/PythonTools/Intellisense/LocationTracker.cs

```diff
@@ -33,7 +33,7 @@ namespace Microsoft.PythonTools.Intellisense {
     /// version of the buffer.
     /// </summary>
     internal class LocationTracker {
-        private ITextSnapshot _snapshot;
+        private readonly Queue<ITextSnapshot> _snapshots;
```

Comment:
What is this change for? `_snapshots` is never iterated.

Reply:
It's a buffer to extend the lifetime of our cache. We limit it to five entries, and as the items leave the queue, their cache entries are cleaned up.

Previously, the queue was effectively one item long, and we were cleaning up cache entries too quickly. Taken to the other extreme, we would never clean up cache entries. This is a tunable compromise. A true solution would be to lock the cache before sending a request to the analyzer, and then unlock it once we've finished dealing with the request, but I didn't want to touch that much code.